### PR TITLE
refactoring code and other small features

### PR DESCRIPTION
**Pull request by 25Pi25**
my main testing was done on a bot clone, the only cases i haven't tested yet are AI descriptions but I assume from the implementation and workaround testing that it should function the same as before

- put all the interaction-based commands in separate files
- defined types for leaderboards and improved leaderboard sorting functions
- added page scrolling for leaderboards
- added type guards for server-specific configs (emoji-based error reactions and mod channel)

### CODE REVIEW

**Comment by 9vult** [comment]
- Loserboard embed does not update page number text when using buttons
- Same for leaderboard
- not a fan of the #/# in the /rank embed, just the rank should be fine
- misc.ts missing `const fetch = require("node-fetch");`
- please change `../credentials.json` back to `./firebase.json` so i don't have to lol
- add safeguards to the embed handlers
- revert `AI-generated` --> `AI-getd` in the help command

File: src/commands/leaderboard.ts
Comment (25Pi25) [inline src/commands/leaderboard.ts:1]: - Loserboard embed does not update page number text when using buttons
- Same for leaderboard

File: src/commands/rank.ts
Comment (25Pi25) [inline src/commands/rank.ts:1]: - not a fan of the #/# in the /rank embed, just the rank should be fine

File: src/misc/misc.ts
Comment (25Pi25) [inline src/misc/misc.ts:1]: - misc.ts missing `const fetch = require("node-fetch");`
- revert AI-generated --> AI-getd in the help command

File: src/raiha.ts
Comment (25Pi25) [inline src/raiha.ts:1]: - please change `../credentials.json` back to `./firebase.json` so i don't have to lol

File: src/commands/leaderboard.ts
Comment (25Pi25) [inline src/commands/leaderboard.ts:1]: - add safeguards to the embed handlers

File: src/misc/misc.ts
Comment (25Pi25) [inline src/misc/misc.ts:1]: - revert AI-generated --> AI-getd in the help command

**Comment by 25Pi25** [comment]
i added too many changes to all be checked in one pull request, I'll keep this version but work on a new pull request with smaller changes

**Pull request closed without merging**